qp_klp/Protocol.py

```diff
@@ -157,6 +165,11 @@ def generate_sequence_counts(self):
         if 'SeqCountsJob' not in self.skip_steps:
             job.run(callback=self.job_callback)
 
+        # if successful, set self.reports_path
+        self.reports_path = join(self.pipeline.output_path,
+                                 'SeqCountsJob',
+                                 'SeqCounts.csv')
```

Comment:
Is this file consistent in schema with `Demultiplex_Stats.csv` used in the other object for `reports_path`?

Reply:
No Demultiplex_Stats.csv's schema is defined and maintained by Illumina and carries additional stats beyond sequence counts. SeqCounts.csv is defined and maintained by us and just records the total raw-read counts for forward and reverse and the lane number per sample.

Consistency in this case is not an issue since the consumer of the metadata is the metapool module, and it only uses Demultuplex_Stats.csv for its raw-read counts. The equivalent output file for the bcl2fastq is similarly different. Our new method is more accurate in that it counts both the forward and reverse reads while I believe in Demultuplex_Stats.csv we just double the forward count.

Reply:
Isn't it risky, and unexpected, for an attribute common across subclasses to have a different interpretation?